### PR TITLE
feat: add includeSymbolicLinks param to findFilesSync

### PR DESCRIPTION
**Pull request by olehrakwix**
Add `includeSymbolicLinks` parameter to `findFilesSync` function that allows include symbolic links in a search results

### CODE REVIEW

**Review by yurii-ve — APPROVED** [review]

**Comment by AviVahl** [comment]
also, avoid link to files in the tests. create junctions.
links to files require Windows' Developer Mode to be turned on when using a clean Windows 11.

File: packages/utils/src/create-extended-api.ts [hunk 1]
```diff
@@ -119,6 +119,8 @@ export function createExtendedSyncActions(baseFs: IBaseFileSystemSync): IFileSys
         filePaths.push(nodePath);
       } else if (item.isDirectory() && filterDirectory(nodeDesc)) {
         findFilesSync(nodePath, options, filePaths);
+      } else if (item.isSymbolicLink() && options.includeSymbolicLinks && filterFile(nodeDesc)) {
```
[the remark this answers is not in the document]
Comment (olehrakwix) [reply packages/utils/src/create-extended-api.ts]: thanks, improved this logic to also include folder junctions

**Comment by olehrakwix** [comment]
@AviVahl , thanks, changed links in tests to be junctions

**Comment by AviVahl** [comment]
also, you only implemented the change for the sync api. there's also async api, with findFiles

**Comment by olehrakwix** [comment]
@AviVahl , added async API

**Comment by AviVahl** [comment]
If resolving link target one must use path.resolve. Using the (possibly relative) link value is wrong, and would break when the link isn't inside cwd (most times).

Let's discuss this when I get back. The entire idea of this function is to find files. It should probably not care about links. It also makes no sense for it to iterate into links to folders (unsafely if I may add).

Just so I'll understand it better... What is it you are trying to achieve with this API change?

**Pull request closed without merging**